noir-projects/noir-protocol-circuits/crates/types/src/hash.nr

```diff
@@ -264,65 +266,50 @@ pub fn poseidon2_hash_with_separator<let N: u32, T>(inputs: [Field; N], separato
 where
     T: ToField,
 {
-    // We manually hash the inputs here, since we cannot express with the type system a constant size inputs array of N + 1
```

Comment:
Seems like the only reason you did this was because we didn't support arithmetic generics at the time you wrote these functions. You should be able to just do `Poseidon2::hash([separator, ...inputs], N+1)` (no spread syntax though).